documentation/bootstrap_upgrade_notes.markdown

```diff
@@ -6,7 +6,25 @@ canonical: "/puppetserver/latest/bootstrap_upgrade_notes.html"
 
 [`bootstrap.cfg` file]: https://docs.puppet.com/puppetserver/2.4/external_ca_configuration.html#disabling-the-internal-puppet-ca-service
 
-> **Note:** If you have disabled the certificate authority (CA) services on a Puppet Server master running Puppet Server 2.4.x or earlier, please read this before upgrading the master to Puppet Server 2.5.0 or newer. If you haven't, or you don't manage or modify your [`bootstrap.cfg` file][], consult the rest of the [Puppet Server 2.5.0 release notes](https://docs.puppet.com/puppetserver/2.5/release_notes.html).
+> ### Potential breaking issues when upgrading with a modified `bootstrap.cfg`
+>
+> If you disabled the certificate authority (CA) on Puppet Server by editing the [`bootstrap.cfg` file][] file on older versions of Puppet Server --- for instance, because you have a multi-master configuration with the default CA disabled on some masters, or use an external CA --- be aware that Puppet Server 2.5.0 no longer uses the `bootstrap.cfg` file.
+>
+> Puppet Server 2.5.0 instead creates a new configuration file, `/etc/puppetlabs/puppetserver/services.d/ca.cfg`, if it doesn't already exist, and this new file enables CA services by default.
+>
+> To ensure that CA services remain disabled after upgrading, create the `/etc/puppetlabs/puppetserver/services.d/ca.cfg` file with contents that disable the CA services _before_ you upgrade to Server 2.5.0. The `puppetserver` service restarts after the upgrade if the service is running before the upgrade, and the service restart also reloads the new `ca.cfg` file.
+>
+> Also, back up your masters' [`ssldir`](https://docs.puppet.com/puppet/latest/reference/dirs_ssldir.html) (or at least your `crl.pem` file) _before_ you upgrade to ensure that you can restore your previous certificates and certificate revocation list, so you can restore them in case any mistakes or failures to disable the CA services in `ca.cfg` lead to a master unexpectedly enabling CA services and overwriting them.
+
+> ### Potential warnings when upgrading with a modified init configuration
+>
+> If you modified the init configuration file --- for instance, to [configure Puppet Server's JVM memory allocation](./install_from_packages.html#memory-allocation) or [maximum heap size](./tuning_guide.html) --- you might see a warning during the upgrade that the updated package will overwrite the file (`/etc/sysconfig/puppetserver` in Red Hat and derivatives, or `/etc/default/puppetserver` in Debian-based systems).
+>
+> The changes to the file support the new service bootstrapping behaviors. If you don't accept changes to the file, you might see a `Service ':PoolManagerService' not found` or similar warning. To satisfy this warning, make sure the `BOOTSTRAP_CONFIG` setting in the init configuration file is set to:
```

Comment:
Technically, this is an `error` rather than a `warning` since it causes the service to fail to start up.